CK3toEU4/Source/EU4World/Output/outNationalIdeas.cpp

```diff
@@ -0,0 +1,34 @@
+#include "outNationalIdeas.h"
+#include "OSCompatibilityLayer.h"
+
+std::ostream& EU4::operator<<(std::ostream& output, const NationalIdeas& idea)
+{
+	output << idea.getDynamicName() + "_ideas = {";
+
+	// National Traditions
+	output << "\n\tstart = {";
+	for (auto& effect: idea.getEthosEffects())
+		output << "\n\t\t " + effect.type + " = " + effect.value;
+	output << "\n\t}";
+
+	// Trigger
+	output << "\n\ttrigger = {\n\t\tculture = " + idea.getDynamicName() + "\n\t}\n";
+	output << "\tfree = yes\n\n";
+
+	// First 7 traditions become ideas, last one becomes ambition
+	for (int i = 0; i < idea.getTraditionIdeas().size() - 1; i++)
```

Comment:
It might be a little more readable (and efficient, though that doesn't really matter) to have 
```
const auto& traditionIdeas = idea.getTraditionIdeas();
```
before the loop and then just use traditionIdeas after.